File: python/collatz-conjecture/collatz_conjecture.py
def steps(number):
    if number <= 0:
        raise ValueError("Only positive integers are allowed")
    amount_of_steps = 0
    while number != 1:
        if number % 2 == 0:
            number /= 2
        else:
            number = 3 * number + 1
        amount_of_steps += 1
    return amount_of_steps

File: lua/collatz-conjecture/collatz-conjecture.lua
return function(n)
  assert(n > 0, "Only positive numbers are allowed")

  function collatz(n)
    if n == 1 then return 0 end

    if n % 2 == 0 then return 1 + collatz(n / 2) end

    return 1 + collatz(3 * n + 1)
  end

  return collatz(n)
end

